### ISSUE TITLE
[feature]: ask password quickly after start

### ISSUE THREAD

**Issue by starius** [lightningnetwork/lnd#7749]
**Is your feature request related to a problem? Please describe.**

Every time I update the node, I have to remember to enter the password after all the databases were fully opened. Since opening sometimes takes more than 10 minutes, I switch to other tasks and eventually forget to do it. Remembering after few hours or because peers start pinging me in messenger "your node is down".

**Describe the solution you'd like**

Ask the password right after the start of the binary, before time consuming procedures.

**Comment by eliotcougar** [comment]
I confirm I was annoyed by this more than once, but because I don't restart the node too often I forget about that annoyance...
Basically, if you try lncli unlock before the database is finished compacting, it silently fails, and the node stays locked. It should take the password regardless of the state, and then wait until all the procedures are complete, then unlock the node...

**Comment by guggero** [comment]
Since we now have a status RPC that is up for the whole duration, this could be done quite easily in `lncli` with a wait loop similar as the one implemented in `lndinit`: https://github.com/lightninglabs/lndinit/blob/main/cmd_wait_ready.go#L70

Instead of waiting for `lnrpc.WalletState_SERVER_ACTIVE` we would wait for `lnrpc.WalletState_LOCKED` until the password is sent to the actual `lnd` wallet unlocker service.

**Comment by AbhinavMir** [comment]
Hi, would love to take this on! Working on this over the weekend. :) This is my first bitcoin ecosystem issue, will probably have some stupid questions.

**Comment by AbhinavMir** [comment]
Thanks! Btw the developer slack link is no longer active, it seems.

**Comment by guggero** [comment]
Ah, it expires every few weeks, we'll update it, thanks for the heads up.

**Comment by eliotcougar** [comment]
> I was looking into this, as I am trying to work on my first beg issue, and I found that an auto-unlock feature already exists in the latest version

That's not very relevant to this issue… I don't want to store my password in a text file on the machine where lnd is running…

**Comment by anasosadances** [comment]
Yeah, true. Ok, I'm gonna try to tackle this :p first issue (I am a beg)

**Comment by Raspiblizz** [comment]
If incorporating a wait loop, would this be handled in the main.go lncli or main.go lnd file? I am thinking it would make more sense to put in the main.go lnd file, but I wanted to double check. Thanx in advance !

**Comment by starius** [comment]
@Raspiblizz If I understand @guggero 's [comment](https://github.com/lightningnetwork/lnd/issues/7749#issuecomment-1594265392) correctly, he is talking about a loop inside `lncli`. The proposed loop code will send RPC to LND to figure out if lnrpc.WalletState_LOCKED is reached. And send unlock RPC after that. Correct me, if I am wrong.

**Comment by srikanth-iyengar** [comment]
I would like to work on this issue, is this still up for Grabs ?

**Comment by malsomesh9** [comment]
is it still open. I would like to work on this issue.

**Comment by eliotcougar** [comment]
> is it still open. I would like to work on this issue.

As far as I can tell, the issue is still there… i.e., if you enter the password while lnd still compresses the database during startup, the password is accepted, but the node isn't unlocked afterwards when the process finishes…

**Comment by starius** [comment]
I fixed this in PR https://github.com/lightningnetwork/lnd/pull/10536
Tested locally - it works.
